src/dataset.ts

```diff
@@ -344,6 +362,13 @@ function checkProperties(
         if ('units' in property && typeof property.units !== 'string') {
             throw Error(`'properties['${key}'].units' should contain a string`);
         }
+
+        // check if parameter keyword exist for multidimensional properties
+        if (Array.isArray(property.values[0])) {
+            if (typeof property.parameter !== 'string') {
+                throw Error(`'properties['${key}'].parameter' should contain a string`);
+            }
+        }
```

Comment:
This should also check that each sub-array have the same size, and that this size matches the size of the parameter.